dashboard/app.py

```diff
@@ -1,55 +1,48 @@
 from concurrent.futures.thread import ThreadPoolExecutor
 
-import yaml
 import os
 import logging
+import importlib
 
-from flask import Flask, session, redirect
+from flask import Flask, session, redirect, request, url_for
 from flask_sslify import SSLify
 from werkzeug.debug import DebuggedApplication
 from authlib.flask.client import OAuth
 from loginpass import create_flask_blueprint, Google
+from werkzeug.contrib.cache import SimpleCache
 
 from dashboard.blueprints.page import page
-from dashboard.blueprints.extra import extra
 from dashboard.dataproviders import *
 from dashboard.service.mysql import MySQLClient
 from dashboard.model.influxdb_data import InfluxDBData
 from dashboard.model.prometheus_data import PrometheusData
 from dashboard.model.tables_data import TableDataProvider
-from dashboard.model.reports_data import ReportsDataProvider
-from dashboard.model.description_data import DescriptionData
 from dashboard.models import ExtraEtl
+from dashboard.utils import get_yaml_file_content
 
-
-TABLES_PATH = '../config/tables.yaml'
-EXTRA_ETL_PATH = '../config/extra_etl.yaml'
-REPORTS_PATH = '../config/reports.yaml'
-
-
-def get_yaml_file_content(path):
-    config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), path)
-    if os.path.exists(config_file):
-        with open(config_file, 'r') as file:
-            return yaml.load(file)
-    else:
-        return None
-
+TABLES_PATH = 'config/tables.yaml'
 
 def setup_authentication(app):
     oauth = OAuth(app)
 
-    def handle_authorize(remote, token, user_info):
+    def handle_redirects(remote, token, user_info):
         if not 'hd' in user_info or user_info['hd'] != app.config['OAUTH_DOMAIN']:
-            return redirect('/login')
+            return redirect(url_for('page.login'))
```

Comment:
How does `url_for` find proper endpoint here?

Reply:
With the blueprint name `page` before the dot.